### PR TITLE
hearing.date / hearing_day.hearing_date -> scheduled_for

### PR DESCRIPTION
**Pull request by sharonwarner**
Connects #8586 

### Description
This PR changes the date field on the hearing object and the hearing_date field on the hearing_day to scheduled_for on both AMA and legacy hearings.

It also adds some of the logic required to pull in AMA hearings on the upcoming hearings page in hearing prep. That will be continued in a future PR.

### Testing Plan
1. Confirm hearings are displayed correctly on the Case Details page
1. Confirm RO and judge algorithms run correctly
1. Confirm scheduling hearings works correctly

I merged a separate PR to make this migration backwards compatible.

### CODE REVIEW

**Comment by cmgiven** [comment]
This looks right to me from an appeals status perspective. Just want to confirm, `scheduled_for` continues to be the correct place to find the date after the hearing has been held (or missed, or what-have-you)? If so, all good here.

**Comment by sharonwarner** [comment]
@cmgiven That's correct!

File: spec/models/hearing_docket_spec.rb [hunk 1]
```diff
@@ -28,11 +28,14 @@
     subject { HearingDocket.from_hearings(hearings) }
 
     let(:hearings) do
-      [Generators::LegacyHearing.create(date: 5.minutes.ago), Generators::LegacyHearing.create(date: 10.minutes.ago)]
+      [
```
Comment (Sjones352) [inline spec/models/hearing_docket_spec.rb]: Passes

File: spec/models/hearing_docket_spec.rb [hunk 2]
```diff
@@ -49,7 +52,7 @@
 
     it "returns a hash" do
       expect(subject.class).to eq(Hash)
-      expect(subject[:date]).to eq(docket.date)
+      expect(subject[:scheduled_for]).to eq(docket.scheduled_for)
```
Comment (Sjones352) [inline spec/models/hearing_docket_spec.rb]: Passes

File: spec/models/hearing_docket_spec.rb [hunk 3]
```diff
@@ -28,11 +28,14 @@
...
-      [Generators::LegacyHearing.create(date: 5.minutes.ago), Generators::LegacyHearing.create(date: 10.minutes.ago)]
+      [
+        Generators::LegacyHearing.create(scheduled_for: 5.minutes.ago),
+        Generators::LegacyHearing.create(scheduled_for: 10.minutes.ago)
+      ]
     end
 
     it "returns the earliest date" do
-      expect(subject.date).to eq 10.minutes.ago
+      expect(subject.scheduled_for).to eq 10.minutes.ago
```
Comment (Sjones352) [inline spec/models/hearing_docket_spec.rb]: Passes

File: spec/models/judge_schedule_period_spec.rb [hunk 4]
```diff
@@ -104,10 +104,20 @@
 
   context "A judge with a lot of non-availability days still gets as many hearings as possible" do
     let!(:hearing_days) do
```
Comment (Sjones352) [inline spec/models/judge_schedule_period_spec.rb]: Passes

File: spec/models/judge_spec.rb [hunk 5]
```diff
@@ -8,10 +8,12 @@
     subject { Judge.new(user).upcoming_dockets }
 
     let(:user) { Generators::User.create }
-    let!(:hearing)            { create(:legacy_hearing, user: user, date: 1.day.from_now) }
-    let!(:hearing_same_date)  { create(:legacy_hearing, user: user, date: 1.day.from_now + 2.hours) }
-    let!(:hearing_later_date) { create(:legacy_hearing, user: user, date: 3.days.from_now) }
-    let!(:hearing_another_judge) { create(:legacy_hearing, user: Generators::User.create, date: 2.days.from_now) }
+    let!(:hearing)            { create(:legacy_hearing, user: user, scheduled_for: 1.day.from_now) }
```
Comment (Sjones352) [inline spec/models/judge_spec.rb]: Passes

File: spec/models/legacy_hearing_spec.rb [hunk 6]
```diff
@@ -7,7 +7,7 @@
   let(:hearing) do
     build(
       :legacy_hearing,
-      date: date,
+      scheduled_for: scheduled_for,
```
Comment (Sjones352) [inline spec/models/legacy_hearing_spec.rb]: Passes

**Comment by Sjones352** [comment]
Hey, Sharon, I have run the tests and they all pass. The tests that were a one-line change were not run with `ensure_stable` tag and the rest with `ensure_stable` tag. Hopefully, I didn't miss anything but I am sure I have covered all tests.

File: app/mappers/hearing_day_mapper.rb [hunk 7]
```diff
@@ -5,7 +5,7 @@ class InvalidRegionalOfficeError < StandardError
   COLUMN_NAME_REVERSE_MAP = {
     hearing_pkseq: :id,
     hearing_type: :hearing_type,
-    hearing_date: :hearing_date,
+    hearing_date: :scheduled_for,
```
[the remark this answers is not in the document]
Comment (sharonwarner) [reply app/mappers/hearing_day_mapper.rb]: In VACOLS, yes. In Caseflow, it'll be scheduled_for.

**Pull request merged**